### ISSUE TITLE
Known iOS 16, watchOS 9 issues

### ISSUE THREAD

**Issue by zacwest** [home-assistant/iOS#2148]
# Notification service goes away

iOS 16 introduces a new privacy feature which disallows the app from knowing your device name. This means your notify target for all devices on iOS 16 is renamed to either `notify.mobile_app_iphone` or `notify.mobile_app_ipad`.

**Workaround**: Rename your device in the app's Settings > [Server] > Device Name.

# Local push does not work

iOS 16 has a crash in the Apple-provided support for local push, causing the Local Push provider to crash when it starts. Waiting to hear back from Apple on resolution.

**Workaround**: None.

**Comment by Smiggel** [comment]
Is this also the reason with notify.<device name> is never registered in HomeAssistant?

**Comment by yan14** [comment]
i just wanted to add #2201 here

**Comment by apercrcl01** [comment]
I have upgraded to iOS 16 and watchOS 9.  I am experiencing issues with apple watch complications.  The only option I have in the watch app is Placeholder.  If I go to watch faces I have already set up they display the names of the complication that I had previously configured.  However, on the watch it says "Not configured - A template for display has not been configured".  I tried recreating one of my complications, but 24 hours later I still only have Placeholder as an option. I have tried various tips and tricks to get the watch app to recognize new complications, but so far no luck.  Is anyone else experiencing this?  Any assistance would be greatly appreciated.  Thank you.

**Comment by toddrob99** [comment]
> # Notification service goes away
> iOS 16 introduces a new privacy feature which disallows the app from knowing your device name. This means your notify target for all devices on iOS 16 is renamed to either `notify.mobile_app_iphone` or `notify.mobile_app_ipad`.
> 
> **Workaround**: Rename your device in the app's Settings > [Server] > Device Name.

The workaround of defining Device Name in the HA app settings appears to work, because I see my `notify.mobile_app_todd_iphone` service in HA. 

However, after I close the HA app on my phone, the `notify.mobile_app_todd_iphone` service disappears and `notify.mobile_app_iphone` shows up again. When I open HA on my phone again, the `notify.mobile_app_todd_iphone` service comes back. Sometimes it seems to keep the named service longer, but other times it reverts after the app has been closed for like 10 seconds. I have multiple iPhones that need to be differentiated, so I can't just use `notify.mobile_app_iphone`.

**Comment by dlindnegm** [comment]
Same here - once you log out of the IOS app and back in it doesn't preserve the Device Name you manually entered and is sending that to HA, which reverts all of the entities back to the generic "sensor.iphone_xxxx".

**Comment by toddrob99** [comment]
It's worse for me than reverting when I log out of the app. While sitting at my computer, I set my phone down on the desk, locked, HA app closed. In HA on desktop I open Developer Tools > Services and type `notify.mobile` in the `Service` field so it shows a list of all iphone and ipad notification services. I leave it like that and just watch, and after some random amount of time (less than a minute usually), my named `notify.mobile_app_todd_iphone` service disappears and the generic `notify.mobile_app_iphone` service appears. Then after a little more time it switches back. I've left it open while working on another screen and it has been switching back and forth, with my phone not being touched.

**Comment by dlindnegm** [comment]
Let me know how that works out for you.  I was also seeing multiple servers in the app at times and would delete them, but couldn't identify the cause.  As part of my testing I've removed the external URL (Duckdns) from all 3 iPhones as well as from the Network section of the HA, and it continues to revert all3 iphones the generic "iPhone" as soon as any one of them logs out of the app and back in.  These issues ([url](https://github.com/home-assistant/iOS/issues/2203)) still have not been assigned to anyone, and the issues was identified 3 months ago.

**Comment by toddrob99** [comment]
I have deleted the second server a dozen times and it keeps coming back... and when it comes back, the Device Name field is blank/default. This time I am setting the Device Name to Todd iPhone and *not* deleting the server. Maybe it will keep the custom Device Name and stop the flapping in HA.

**Comment by zacwest** [comment]
Hi folks, I'm aware of the notification service name issue (we requested but never received from Apple approval to access device names well in advance and I was counting on approval; it honestly slipped my mind until release day) but unfortunately haven't had a chance to implement a workaround. I hope to get this fixed and out in the next week or two by prompting for a device name whenever we do not know one -- both when setting up a server and for already-set-up servers.

The one issue with the workaround in the top of the issue is `sourceDeviceID` is incorrect in events from the device; this was an oversight when I implemented device name overriding.

I'm sorry for the inconvenience but please be patient as I've just welcomed my daughter into the world this weekend and have been otherwise occupied recently.

**Comment by ntompson** [comment]
I'm sorry for the dumb question, but I'm not quite clear. Are complications completely broken in watchOS 9, or is it just that they can't be added / edited?

I've upgraded my phone to iOS 16, but I use quite a few watch complications, so I'm holding off on the watchOS upgrade. So I'm wondering whether I should continue to hold off. 

If complications are completely broken, may I suggest creating / pinning a new issue just to cover the watchOS complication problem?

**Comment by ntompson** [comment]
Grrr. My watch go updated to watchOS 9 overnight - I had forgotten auto-updates were on. But the good news is that complications seem to be working. Is this expected?

**Comment by alotofbits** [comment]
> I have deleted the second server a dozen times and it keeps coming back... and when it comes back, the Device Name field is blank/default. This time I am setting the Device Name to Todd iPhone and _not_ deleting the server. Maybe it will keep the custom Device Name and stop the flapping in HA.

I have a variation of the workaround that seems to work - at least until the app forgets that it has a ser er defined:
1. Log in with the app and set a custom devicename
2. Delete the app 
3. Delete the device in the HA mobile app-integration
4. Reinstall the app and and go through all the permission-dialogs. Somehow the app remebers the previous server definition and reconnects to to server using  the custom devicename.
5. Wait for the app to forget the server - which seems to happend once every few days - and repeat from 1.

Hope this helps someone…

**Comment by salleq** [comment]
> Grrr. My watch go updated to watchOS 9 overnight - I had forgotten auto-updates were on. But the good news is that complications seem to be working. Is this expected?

I don't know if this will help you, but at least one complication is working for me:
Creating "Round" complication for "Graphical" Watchface works at least for some parts of "Infographic",  "Modular", "Meridian" watchfaces. Hope I translated them correctly.

**Comment by Reachushere** [comment]
> Hi folks, I'm aware of the notification service name issue (we requested but never received from Apple approval to access device names well in advance and I was counting on approval; it honestly slipped my mind until release day) but unfortunately haven't had a chance to implement a workaround. I hope to get this fixed and out in the next week or two by prompting for a device name whenever we do not know one -- both when setting up a server and for already-set-up servers.
> 
> The one issue with the workaround in the top of the issue is `sourceDeviceID` is incorrect in events from the device; this was an oversight when I implemented device name overriding.
> 
> I'm sorry for the inconvenience but please be patient as I've just welcomed my daughter into the world this weekend and have been otherwise occupied recently.

Congrats!  You are forgiven :)

**Comment by SHxKM** [comment]
For me, HomeKit’s lock widget is too-often showing “All Off” even when that’s practically never the case in my house.

**Comment by ntompson** [comment]
> Edit: I should've read your message better. Which complications do you have working? Can you screenshot?

These are working fine:

![incoming-F383A47B-C430-42ED-AD09-0AA28A9C1177](https://user-images.githubusercontent.com/32885870/191899508-16f48fdd-f3c1-4632-b2a3-14d62d580974.PNG)

All but the bottom right are HA complications.

**Comment by matteocorti** [comment]
@ntompson: can you please post a screenshot of the configuration?

**Comment by yan14** [comment]
> configuration?

I had to install my watch again (switch from old iPhone to new). 
To get my complications back, I had to log out of my Wi-Fi and do a manual update in the HA App.
Afterwards everything was back and worked as befor!

**Comment by werehamster999** [comment]
@zacwest 

> I'm sorry for the inconvenience but please be patient as I've just welcomed my daughter into the world this weekend and have been otherwise occupied recently.

A hearty congratulations from my family to yours. A programmer AND a daddy? I see copious amounts of coffee in your future!
All the best!
Keith

**Comment by Staffman** [comment]
> > configuration?
> 
> I had to install my watch again (switch from old iPhone to new). To get my complications back, I had to log out of my Wi-Fi and do a manual update in the HA App. Afterwards everything was back and worked as befor!

"I had to log out of my Wi-Fi and do a manual update in the HA App."
I didn't catch this. Can you describe what to do please :)

**Comment by yan14** [comment]
> > > configuration?
> > 
> > 
> > I had to install my watch again (switch from old iPhone to new). To get my complications back, I had to log out of my Wi-Fi and do a manual update in the HA App. Afterwards everything was back and worked as befor!
> 
> "I had to log out of my Wi-Fi and do a manual update in the HA App." I didn't catch this. Can you describe what to do please :)

Disable Wi-Fi, then open the HA App, Go to settings --> mobile App --> Apple Watch and click on the Blue Field under manual Updates (German "Fehler beim Update"). Afterwards everything was back on my watch. Now you can Enable Wi-Fi again!

**Comment by automationfixation** [comment]
Hi, just to add to this, I've played around with the new watchOS 9 and Complications and worked out the ones that are working and not working. On Apple's Complication Guide here: https://developer.apple.com/design/human-interface-guidelines/components/system-experiences/complications they have referred to a number of templates as 'Legacy' which I assume means they are phasing these out or the update may have broken the app's config? These 'Legacy' templates appear to be the ones that are no longer working.

The ones that are working are in line with the Apple Guide above and are Graphic Circular, Graphic Corner, Graphic Rectangular and Modular Large. I've tried these out on watchOS 9 and iOS 16 and they work fine. I have put the steps taken here in this video: https://www.youtube.com/watch?v=7t22v6Wh6HM

**Comment by K4CZP3R** [comment]
None of the widgets are working on my watch, I'm running iOS 16 & WatchOS 9.

**Comment by alotofbits** [comment]
> > Just checking back for an update on the issue and can confirm that the workaround for renaming of the iPhone notification service still does not work.  The companion app and the entity in HA continue to revert to the generic "iphone" name which breaks any notifications sent to the device.  Same for the second workaround for push notifications - deleting and reinstalling the app (as well as deleting the device in HA) does not allow the correct device name to 'stick'.  Hoping this get's fixed soon!

Check my post above from Sep 21 - it seems to work reliably for me and my family and a few friends.

**Comment by dlindnegm** [comment]
> > > Just checking back for an update on the issue and can confirm that the workaround for renaming of the iPhone notification service still does not work.  The companion app and the entity in HA continue to revert to the generic "iphone" name which breaks any notifications sent to the device.  Same for the second workaround for push notifications - deleting and reinstalling the app (as well as deleting the device in HA) does not allow the correct device name to 'stick'.  Hoping this get's fixed soon!
> 
> Check my post above from Sep 21 - it seems to work reliably for me and my family and a few friends.

I'll give this a try again. Not sure I had waited for it to 'forget' the server and repeated the process a second time.  At one point I had tested creating a second server entry, one for local/home and one for external (DuckDNS), and even though I've deleted each of those I often find they re-appear in the companion app.  Seems like things work up until I either log out of the companion app or leave my Home zone, at which point I either get multiple server entries and/or the generic "iPhone" entry in the Integration.  Crossing my fingers....

**Comment by mww012** [comment]
I've not been able to keep the workaround going either. At one point the I was good for about 24 hours. After a while the entries just show back up and my notification service no longer works.

What I found was if I completely close the app (swipe up from the bottom, swipe up on the HA card) all the entries come back when I open the app back up immediately.

**Comment by dlindnegm** [comment]
Yea, I'm not sure what "wait for the app to forget the server'" means either.  I'm not sure I've ever had it not show a server entry, just duplicates.  After completing steps 1-4, I left my Wi-Fi and when I returned my cellular IP was banned from too many bad login attempts, even though my external URL was correctly picked up from HA when I first signed in after deleting the companion app.  When I checked the Integration it has once again reverted back to the generic "iPhone".  So maybe this works as long as you never log out of the iphone app, never restart your iphone, and never switch from the internal to external URL.

**Comment by alotofbits** [comment]
> Yea, I'm not sure what "wait for the app to forget the server'" means either. I'm not sure I've ever had it not show a server entry, just duplicates. After completing steps 1-4, I left my Wi-Fi and when I returned my cellular IP was banned from too many bad login attempts, even though my external URL was correctly picked up from HA when I first signed in after deleting the companion app. When I checked the Integration it has once again reverted back to the generic "iPhone". So maybe this works as long as you never log out of the iphone app, never restart your iphone, and never switch from the internal to external URL.

Hopefully step 5 never occurs - my setup has been stable now for the past couple of weeks. I have restarted my Iphones, upgraded my Iphones and I use Nabu Casa for remote access.

In step 4 you should not set up a new server connection. It should already be there you just have to approve the apps access to location, notifications etc. Do you use Icloud for backups etc?

**Comment by toddrob99** [comment]
> > So maybe this works as long as you never log out of the iphone app, never restart your iphone, and never switch from the internal to external URL.
> 
> I'm going to leave it broken until fixed I think. Had to disable all my notifications to phones which is a bummer.... Maybe this will make me find a better notification route. :(

I took the opportunity to point all my notifications to a telegram group. That way I can see history, which is nice. It doesn't have actionable push messages, but it does support buttons along with the message in the group/chat itself (have to open the app from the push message).

**Comment by dlindnegm** [comment]
No choice but to wait for an update.  My workarounds are either sending an SMS message via e-mail to my Verizon vtext, or just adding notify.mobile_app_iphone to all of my automations.  The notify method is only good if you want everyone to receive the same message since you never know what the iphones are currently named.  Thanks for the Telegram reminder - I set it up once-upon-a-time and haven't done much with it.   The group option looks like it will come in handy.

**Comment by jum0n** [comment]
> I took the opportunity to point all my notifications to a telegram group. That way I can see history, which is nice. It doesn't have actionable push messages, but it does support buttons along with the message in the group/chat itself (have to open the app from the push message).

Thanks for the tip!  I've set this up and its working great so far.  Also, it gets me history that I wanted, as you stated.  Not the 'easiest' setup, but what is?  So glad I moved all my notifications to scripts and I only have to edit this in a few places instead of in each automation that has one.

**Comment by zacwest** [comment]
iOS app version 2022.10 is now out [in beta](https://home-assistant.io/ios/beta) and should be rolling out in release over the next few days. This includes fixes for device name handling, but does not address anything with Watch Complications, which will require significantly more time investment.

**Comment by mww012** [comment]
Awesome!  Thanks for all your work on this!  You guys have done a great job and I really appreciate what you do.

**Comment by zacwest** [comment]
Discovered an issue in the fix in 2022.10 -- if local push is set up and working, it will break the name again as the local push extension does not have device name access but it is telling the server its registration info when it should not. You can temporarily disable Local Push for your server in the app's settings until a future app release.

**Comment by dlindnegm** [comment]
Not sure if this is also a bug, but if you log out of the app and log back in it re-enables local push.  I need to do some testing to see if changing geo-locations does the same.  The good news is that restarting the iPhone doesn't seem to re-enable it!

**Comment by zacwest** [comment]
The horror of this name redaction continue on macOS 13. While we were granted access to the device name on iOS, it was not granted on Mac (perhaps they forgot they were redacting it there) and so Macs will all become "iPad" upon upgrade.

**Comment by mww012** [comment]
So what I'm hearing is that if I've enjoyed this crazy Apple app ride on my iPhone I should immediately upgrade my Mac? /s

But seriously, thanks for the heads up and all your work on this!  I'll hold off on that app upgrade for a bit.

**Comment by deviantintegral** [comment]
> so Macs will all become "iPad" upon upgrade.

So this is where the long rumoured deprecation of macOS begins 😂 .

> You can temporarily set a device name manually in the server's settings

Does this still work as a workaround for macOS?

**Comment by zacwest** [comment]
2022.10.1 is now out which should resolve the last of the issues for iOS.

**Comment by chumbazoid** [comment]
iOS 16.1: Whenever the phone is rebooted, Local Push apparently reverts to Disabled.

Upon reboot: The LP switch under Internal URL remains active, but LP in Server Status shows as disabled. I have to turn the switch off and save, turn it back on and save again, then close and restart the app before LP shows as Available in Server Status.

**Comment by gabfv** [comment]
> I have upgraded to iOS 16 and watchOS 9. I am experiencing issues with apple watch complications. The only option I have in the watch app is Placeholder. If I go to watch faces I have already set up they display the names of the complication that I had previously configured. However, on the watch it says "Not configured - A template for display has not been configured". I tried recreating one of my complications, but 24 hours later I still only have Placeholder as an option. I have tried various tips and tricks to get the watch app to recognize new complications, but so far no luck. Is anyone else experiencing this? Any assistance would be greatly appreciated. Thank you.

I get the same issue and it has happened for quite a while. However, some complications are shown but most do not show in the Watch app. It also happened with iOS 15 and watchOS 8 in the past, but now I am on the latest for both and the issue still persist.

**Comment by ntompson** [comment]
Just wondering where watch complications are up to. 

I'm finding that my complications do not update under iOS 16 / watchOS 9 - which I think is expected. But if I push a manual update from with the iOS companion app settings, the complications **do** update. So I thought maybe I could implement a workaround by pushing manual updates using notifications:

```yaml
automation:
  - alias: Periodic Apple Watch update
    trigger: 
      - platform: time_pattern
        minutes: "/30"
    action:
      - service: notify.mobile_app_nicks_iphone
        data:
          message: update_complications
```

However, this doesn't do anything. I have validated that normal notifications to `notify.mobile_app_nicks_iphone` works.

So... is this expected? It seems odd to me that updates pushed from the app settings works, but automated updates do not - even if the underlying watchOS 9 problem hasn't been fixed yet.

**Comment by jmwhite5** [comment]
> Does this still work as a workaround for macOS?

After manually overriding the macOS app name, the sensor notifications still carry the wrong name (iPad).  So the workaround doesn't fully work for MacOS.

**Comment by krissen** [comment]
Adding #1925. (#2037 is a dupe.)

This has been a problem since iOS 15, watchOS 8. 

Curiously, the issue initially plagued several apps, not only Home assistant companion. However, at the present moment, the issue has been fixed in other apps. The only app remaining that I personally use, that constantly keeps asking for permissions, is the Home Assistant Companion. I think it might at least be worth looking into again if the matter could possibly be fixed in the Companion.

**Comment by JonLaliberte** [comment]
> > Does this still work as a workaround for macOS?
> 
> After manually overriding the macOS app name, the sensor notifications still carry the wrong name (iPad). So the workaround doesn't fully work for MacOS.

I had this issue just now. The easiest way to fix it is to:
* Rename in the App settings (and get it to show up in HA such that the Device name is correct, but the sensors are named "iPad").
* Change the device name to iPad and tell it to rename everything.
* Change the device name to whatever you actually want it to be and tell it to rename everything again.
*

**Comment by wernerhp** [comment]
Mac OSX Ventura 13.2
Home Assistant Companion 2023.2 (2023.444)

- Camera status doesn't update reliably.  Seems to only update when I close and re-open the app.  Manually updating doesn't help.
- Audio Input/Output does not register for Bluetooth headphones (Sony WH-1000XM3).  Disconnecting Bluetooth updates as expected.

**Comment by deviantintegral** [comment]
Anyone know of new issues on the iOS 17 / watchOS 10 / macOS 14 releases?

If there's nothing new but the above reported issues still matter, perhaps this issue is just renamed? Or it could be closed in favour of a new one.

**Comment by parautenbach** [comment]
> Mac OSX Ventura 13.2 Home Assistant Companion 2023.2 (2023.444)
> 
> * Camera status doesn't update reliably.  Seems to only update when I close and re-open the app.  Manually updating doesn't help.
> * Audio Input/Output does not register for Bluetooth headphones (Sony WH-1000XM3).  Disconnecting Bluetooth updates as expected.

Werner, just looking at your comment again. For you and others: I discovered a code snippet some months ago and last week at last implemented a workaround for the BT headphones issue. 

If you're interested:

* [micstatus.py](https://github.com/parautenbach/Home-Automation/blob/master/scripts/micstatus.py) (see the comments in file for reference to the original source)
* [plist file](https://github.com/parautenbach/Home-Automation/blob/master/scripts/net.ov1d1u.pyhton.micpixel.plist) for running this as a service (must be updated with the correct script and Python paths)
* [package](https://github.com/parautenbach/Home-Automation/blob/master/home_assistant/packages/work.yaml) containing my application of this

Basically, I use both the companion app and my sensor (the one suffixed with `_alt`):

```
binary_sensor:
  - platform: template
    sensors:
      on_air:
        # https://github.com/home-assistant/iOS/issues/1903
        value_template: >
          {{ is_state('binary_sensor.gaia_audio_input_in_use', 'on') or
             is_state('input_boolean.gaia_audio_input_in_use_alt', 'on') }}
```
